src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const res = [];
+
+  for (const i of actions) {
+    if (i.type === 'addProperties') {
+      Object.assign(stateClone, i.extraData);
+    }
+
+    if (i.type === 'removeProperties') {
+      for (const j of i.keysToRemove) {
+        delete stateClone[j];
+      }
+    }
+
+    if (i.type === 'clear') {
+      Object.keys(stateClone).forEach((key) => delete stateClone[key]);
+    }
```

Comment:
You are using multiple if statements to handle different types of actions. This is a perfect case for a switch statement, which will improve readability and performance of your code. Each action type ('addProperties', 'removeProperties', 'clear') can be a case in the switch.